RPIPs/RPIP-53.md

```diff
@@ -0,0 +1,162 @@
+---
+rpip: 53
+title: Rewards Tree v10 Specification
+description: Adjusts the rewards file to account for Houston changes, generalizing rewards from nodes to claimers and supporting the RPL withdrawal address
+author: Joe Clapis (@jcrtp), Jacob Shufro (@jshufro)
+discussions-to: https://dao.rocketpool.net/t/rewards-tree-spec-v10/2937
+status: Review
+type: Protocol
+created: 2024-04-19
+---
+
+
+## Abstract
+
+The incoming Houston upgrade introduces [RPIP-31](https://github.com/rocket-pool/RPIPs/blob/main/RPIPs/RPIP-31.md) - the **RPL withdrawal address**. This is a supplementary setting, different from the "regular" withdrawal address used in Atlas and below, that node operators can opt into. It's intended to specify a "secondary" withdrawal address that the node's RPL rewards will be sent to, whereas the node's ETH rewards will be sent to its "regular" withdrawal address (known in Houston as a **primary** withdrawal address).
+
+Realistically, most node operators won't end up using this feature. It's only really helpful for node operators that don't supply their own RPL when creating a minipool. If you don't fall into this camp, rest assured that *this proposal will have no impact on you*.
+
+Unfortunately, node operators that *do* use this new feature can **no longer trigger rewards claims** from the Merkle rewards system. More specifically they can distribute minipool rewards from the Beacon Chain, but can't claim the rewards produced every 28 days via from the Merkle rewards system. That's not a problem for RPL rewards, since the RPL withdrawal address gets those anyway, but the issue is this includes Smoothing Pool ETH rewards as well. Neither the node not the address that would get the Smoothing Pool ETH (the primary withdrawal address) can actually trigger a claim; *only* the RPL withdrawal address will be able to do it.
+
+This can't be fixed for nodes that opt into the RPL withdrawal address set if they also use their node as their primary withdrawal address because of technical reasons. However, it *can* be fixed for nodes that opt into it and use an address *other* than their node as the primary withdrawal address. I anticipate that most nodes opting into the RPL withdrawal address feature will fall into this camp.
```

Comment:
Are these technical reasons ever explained? Is it the note at the bottom about avoiding an extra transaction? I don't think I agree with this decision if so.

Reply:
There was a long discussion about this in #governance - basically the problem is if you don't include this particular condition, there will be two entries created in the tree: one at the node address with the node's ETH, and one at the RPL withdrawal address for the RPL. Since the former is only claimable by the RPL withdrawal address according to the Houston contracts, it means the RPL withdrawal address has to claim twice: once for their RPL, and once for the node's ETH. There's no direct incentive for them to do the latter since it costs them gas and they get nothing out of it, so we decided it was best to keep both sets of rewards in a single entry to ensure the RPL withdrawal address *does* have an incentive to claim.

I ended up taking that out of the spec for brevity but I can add it back in if you think it makes sense to do so.